src/sorted_vector.jl

```diff
@@ -0,0 +1,52 @@
+"""
+A `SortedVector` behaves like a standard Julia `Vector`, except that its elements are stored in sorted order, with an optional function `by` that determines the sorting order in the same way as `Base.searchsorted`.
+"""
+immutable SortedVector{T, F<:Function}
+    data::Vector{T}
+    by::F
+
+    function SortedVector(data::Vector{T}, by::F)
+        new(sort(data), by)
```

Comment:
If the constructor doesn't call `sort`, you can end up with a non-sorted `SortedVector`. Or am I missing something? Or just call `issorted` first to see if it needs sorting? (But is that much faster than just calling `sort` if it's already sorted?)